Go/search/interpolation_search.go

```diff
@@ -0,0 +1,45 @@
+package main
+import "fmt"
+ 
+func Search(array []int, number int) int {
+	
+	low := 0
+	high := len(array)-1
+ 
+    for {
+	var key int
+	var size int
+	
+        if high == low {
+            key = high
+        
+	} else {
+		size = high - low
+		key = low + (int(float64(size-1)*(float64(number-array[0])/float64(array[len(array)-1]-array[0]))))
+        }
+ 
+        if array[key] == number {
+		return key
+        }
+ 
+        if array[key] > number {
+            high = key - 1
+        
+	} else {
+            low = key + 1
+        }
+    }
+}
+
+// Input
+
+func main(){
+    items := []int{1, 16, 28, 37, 49, 52, 60, 75, 85, 99, 105}
```

Comment:
Use Dynamic Input instead of Driver Code

Reply:
I did the rest of the changes but I'm unable to understand how to do this one, can you please guide me?